Web/Edubase.Web.UI/Areas/Groups/Controllers/GroupController.cs

```diff
@@ -845,27 +848,11 @@ private async Task<GroupEditorViewModel> PopulateSelectLists(GroupEditorViewMode
 
         private GroupEditorViewModel SetEditPermissions(GroupEditorViewModel viewModel)
         {
-            viewModel.CanUserCloseAndMarkAsCreatedInError = viewModel.GroupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust, GT.SchoolSponsor, GT.Federation)
-                                                               && !viewModel.StatusId.OneOfThese(GS.CreatedInError, GS.Closed)
-                                                               && User.InRole(AuthorizedRoles.IsAdmin);
-
-            viewModel.IsLocalAuthorityEditable = viewModel.GroupTypeId.OneOfThese(GT.ChildrensCentresCollaboration, GT.ChildrensCentresGroup)
-                                                 && viewModel.LinkedEstablishments.Establishments.Count == 0 && User.InRole(AuthorizedRoles.IsAdmin);
-
-
-            if (User.InRole(AuthorizedRoles.CanBulkAssociateEstabs2Groups) && viewModel.GroupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust))
+            viewModel = GroupEditorViewModelRulesHandler.SetEditPermissions(viewModel, User);
+            if (GroupEditorViewModelRulesHandler.UserCanEditClosedDateAndStatus(viewModel, User))
             {
-                viewModel.CanUserEditClosedDate = true;
-                viewModel.CanUserEditStatus = true;
                 PopulateStatusSelectList(viewModel);
             }
-
-            if (User.InRole(AuthorizedRoles.IsAdmin) &&
-                viewModel.GroupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust))
-            {
-                viewModel.CanUserEditUkprn = true;
-            }
-
```

Comment:
I am appreciating this refactor, as it really helps to separate levels of abstraction! 👍